### DISCUSSION TITLE
Incorrect request.scope['router'] from request object

### DISCUSSION THREAD

**Discussion by adarshpunj** [fastapi/fastapi#8661] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [x] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
router = APIRouter(prefix="/myrouter", tags=['mytag'])

@router.get("/meta", status_code=200)
@api_log
async def endpoint(commons: dict = Depends(common_parameters)):

   rtr = commons['request'].scope['router']
   
   prefix = rtr.prefix
   tags = rtr.tags
   
   print(prefix)
   print(tags)
```

### Description

I have an `APIRouter` router, which I've registered in the main file using `include_router`.
I'm trying to get router tags/prefix from the request object.

The output I expect:
```
/myrouter
['mytag']
```

The output I'm getting:
```

[]
```
Is there any way to get router tags from the request object? 
What am I doing wrong here?

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9.7

**Comment by 0417taehyun** [comment]
I'm a beginner of FastAPI and programming, so this might be wrong. Just feel free to read this.

I think the codes below can be the simple solution.

``` Python
# main.py

from fastapi import FastAPI

from api import router

app = FastAPI()

app.include_router(router=router)
```


```Python
# app/__init__.py

from fastapi import APIRouter

from api import endpoint, test

router = APIRouter()

router.include_router(
    router = endpoint.router
)

router.include_router(
    router = test.router
)

```

```Python
# app/test.py
from fastapi import APIRouter


router = APIRouter(prefix="/test", tags=['test'])

@router.get("", status_code=200)
async def test():
    print(router.tags) # > ['test']
```

```Python
# app/endpoint.py
from fastapi import APIRouter


router = APIRouter(prefix="/endpoint", tags=['endpoint'])

@router.get("", status_code=200)
async def endpoint():
    print(router.tags) # > ['endpoint']
```

If you call the `Request` object directly, and call `router` from it, makes a different object like code below. 
So, if you are busy it is better to solve this problem by calling `router` in each file like codes above.

```Python
from fastapi import APIRouter, Request


router = APIRouter(prefix="/endpoint", tags=['endpoint'])


@router.get("", status_code=200)
async def endpoint(request: Request):
    print("---the origin")
    print(id(router))
    
    print("---by Request")
    print(id(request.app.router)) # > Creating another object.
```

In docs, [Request](https://fastapi.tiangolo.com/advanced/using-request-directly/#details-about-the-request-object), it says:

> It would also mean that if you get data from the Request object directly (for example, read the body) it won't be validated, converted or documented (with OpenAPI, for the automatic API user interface) by FastAPI.

the point is **won't be ... documented ... by FastAPI** and I think `tags` are information for OpenAPI, that's why `Request` can't catch it.

I will keep trying it. Can you leave a comment if you solve this problem? It would be helpful to me also. Sorry that it might not be a solution.

**Comment by adarshpunj** [comment]
Hi @0417taehyun, thanks much for the reply, and for highlighting that point in the doc. 
We can use `router` but it will be tedious to do so for every single function, and for every router. I'm trying to use a decorator to access info about the router (maybe during logging an API call). 
Let's see if there's a workaround. You can follow this question at [Stack Overflow](https://stackoverflow.com/questions/69700468/fastapi-incorrect-router-in-request-scope).
